networkx/generators/geometric.py

```diff
@@ -844,3 +845,206 @@ def thresholded_random_geometric_graph(
     )
     G.add_edges_from(edges)
     return G
+
+
+@py_random_state(5)
+def soft_random_S1_H2_geometric_graph(
+    beta, *, n=None, gamma=None, mean_degree=None, kappas=None, seed=None
+):
+    r"""Returns a $\mathbb{S}^1/\mathbb{H}^2$ model.
+
+    The $\mathbb{S}^1$ model [1]_ is the geometric soft configuration model
+    which is able to explain many fundamental features of real networks such as
+    small-world property, heteregenous degree distributions, high level of
+    clustering, and self-similarity.
+
+    In the one-dimensional $\mathbb{S}^1$ model, a node $i$ is assigned two hidden
+    variables: a hidden degree $\kappa_i$, quantifying its popularity, influance,
+    or importance, and an angular position $\theta_i$ in a one-dimensional
+    sphere (or circle) abstracting the similarity space, where angular distances
+    between nodes are a proxy for their similarity. The radius of the circle is
+    adjusted to $R = N/2\pi$,  where $N$ is the number of nodes, so that the density
+    is set to 1 without loss of generality.
+
+    The connection probability between any pair of nodes takes the form
+    of a gravity law, whose magnitude increases with the product of the
+    hidden degrees (i.e., their combined popularities), and decreases with
+    the angular distance between the two nodes.
```

Comment:
Maybe something like:
```
    The connection probability pairs of nodes is similar to a gravity law in that
    hidden degree acts like mass (causing attraction) and that attraction decreases
    with angular distance (though only linearly unlike gravity's square dependence).
```
Maybe the gravity bit is a distraction -- explaining the increase and decrease might be enough, but you can decide.